app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,22 @@
+class VotesController < ApplicationController
+  def create
+    @vote = Vote.new(user_id: session[:user_id], work_id: params[:work_id])
+
+    if Vote.find_by(user_id: session[:user_id], work_id: params[:work_id]) != nil
```

Comment:
Notice how complicated this method is.  Putting the business logic into a model method would be appropriate.